lightning/src/routing/network_graph.rs

```diff
@@ -375,17 +375,20 @@ impl<C: Deref , L: Deref > RoutingMessageHandler for NetGraphMsgHandler<C, L> wh
 		let batch_count = batches.len();
 		for (batch_index, batch) in batches.into_iter().enumerate() {
 			// Per spec, the initial first_blocknum needs to be <= the query's first_blocknum and subsequent
-			// must be >= the prior reply. We'll simplify this by using zero since its still spec compliant and
-			// sequence completion is now explicitly.
-			let first_blocknum = 0;
+			// must be >= the prior reply. We'll simplify this by using the query's first_blocknum since its
+			// still spec compliant and sequence completion is now explicit.
+			let first_blocknum = msg.first_blocknum;
 
-			// Per spec, the final end_blocknum needs to be >= the query's end_blocknum, so we'll use the
-			// query's value. Prior batches must use the number of blocks that fit into the message. We'll
-			// base this off the last SCID in the batch since we've somewhat abusing first_blocknum.
+			// Per spec, the last end_block needs to be >= the query's end_blocknum. Last
+			// reply calculates difference between the query's end_blocknum and the start of the reply.
+			// Overflow safe since end_blocknum must be >= msg.first_block_num.
 			let number_of_blocks = if batch_index == batch_count-1 {
```

Comment:
Confused -- is `number_of_blocks` the total number of blocks for the whole sequence of `ReplyChannelRange`'s, or just for this specific one (which may be part of a larger sequence of `ReplyChannelRange`'s)?